### ISSUE TITLE
could vec_as_names argument "quiet" be configurable via options?

### ISSUE THREAD

**Issue by cboettig** [r-lib/vctrs#1547]
Other packages often call `vec_as_names` internally, usually making it impossible to take advantage of this option. (e.g. `readr`).  I think it would be very convenient if the default for `quiet` was something like `quiet = getOption("vctrs.quiet", FALSE)` instead of just FALSE.

**Comment by cboettig** [comment]
Thanks @lionel- , that makes more sense to me too.

**Comment by jennybc** [comment]
Reviewing a PR in readr that addresses @cboettig's issue prompted me to re-orient myself to name repair (https://github.com/tidyverse/readr/pull/1426).

It looks like `vctrs::vec_as_names()` **already does** have exactly what @cboettig  requested. The `rlib_name_repair_verbosity` option allows a user to suppress name repair messaging, even if there are several packages / function calls in the way, i.e. the user doesn't have direct access to the `quiet` argument of `vec_as_names()`.


```
#' @param quiet <blah, blah>
#'
#'   Users can silence the name repair messages by setting the
#'   `"rlib_name_repair_verbosity"` global option to `"quiet"`.
```

And, indeed, it works:

``` r
library(readr)

# let's suppress this noise for the moment, to focus on name repair 
withr::local_options(list(readr.show_col_types = FALSE))

read_csv(I("x,x\n1,2\n3,4"))
#> New names:
#> • `x` -> `x...1`
#> • `x` -> `x...2`
#> # A tibble: 2 × 2
#>   x...1 x...2
#>   <dbl> <dbl>
#> 1     1     2
#> 2     3     4

withr::with_options(list(rlib_name_repair_verbosity = "quiet"),
  read_csv(I("x,x\n1,2\n3,4"))
)
#> # A tibble: 2 × 2
#>   x...1 x...2
#>   <dbl> <dbl>
#> 1     1     2
#> 2     3     4

op <- options(rlib_name_repair_verbosity = "quiet")
read_csv(I("x,x\n1,2\n3,4"))
#> # A tibble: 2 × 2
#>   x...1 x...2
#>   <dbl> <dbl>
#> 1     1     2
#> 2     3     4
options(op)
```

<sup>Created on 2022-08-29 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

To more clearly document what (I think) happens, vctrs delegates name repair messaging to rlang and rlang honors the `rlib_name_repair_verbosity` option.

https://github.com/r-lib/vctrs/blame/main/NEWS.md#L163-L168

https://rlang.r-lib.org/reference/names_inform_repair.html?q=names_inform_#ref-usage

I'm wondering why this didn't come up here.

Is it just a case of forgetting about this option? Or that the option is an rlang matter, not a vctrs one? Or am I not putting the pieces together correctly?

**Comment by lionel-** [comment]
yup I think I just forgot about that option when I closed this issue. I still think it might be a good idea to provide package-specific quieting options, to avoid users just turning it off globally in their RProfile.

Or maybe we should just turn off verbosity by default and provide tools for inspecting backtraces of all events of name repairing during a command, e.g. via `rlang::last_name_repairs()`. Currently even with verbosity turned on it's not always evident for end users where exactly a name repair happened.
